src/components/ModularTable/ModularTable.test.tsx

```diff
@@ -4,10 +4,10 @@ import React from "react";
 import ModularTable from "./ModularTable";
 
 const columns = [
-  { accessor: "status", Header: "Status" },
-  { accessor: "cores", Header: "Cores", className: "u-align--right" },
-  { accessor: "ram", Header: "RAM", className: "u-align--right" },
-  { accessor: "disks", Header: "Disks", className: "u-align--right" },
+  { accessor: "status" as const, Header: "Status" },
```

Comment:
I'm curious as to why this was necessary? Is this something we'll have to do when consuming the modular table in the apps (if so it'll be a breaking change)?

Reply:
It is a breaking change I'm afraid. Previously TypeScript would not complain about an invalid accessor string value. Now - because I'm passing down the generic to react-table types they expect correct value for column accessors (not just a string, but e.g. "status").

Reply:
Happy to open another PR and change this (or anything else) before release if needed.

Reply:
Looping you in @bartaz

Reply:
Thanks. Saw this earlier but didn't have time to have a deeper look to understand the change, will try to have a look today.

I wonder how breaking of a change is it? The API or data structure doesn't seem to change. So is it just that TS compiler will complain?

Would it still work just complain about types? Would it matter at all if project itself uses JS and not TS?

Reply:
I believe the TS build would fail (depending on how strict the config is in each project). JS projects would continue to work just fine.

Reply:
The issue is detailed here with some examples: https://github.com/TanStack/react-table/discussions/2664#discussioncomment-335443

Reply:
@bartaz I opened another PR that removes ModularTable strict type https://github.com/canonical-web-and-design/react-components/pull/733

This will give us a bit more time to think about this (stricter type is not required for the new ModularTable functionality to work).